File: src/main/java/net/atos/zac/admin/model/HumanTaskReferentieTabel.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.admin.model;

import static net.atos.zac.util.FlywayIntegrator.SCHEMA;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import nl.info.zac.admin.model.ReferenceTable;

@Entity
@Table(schema = SCHEMA, name = "humantask_referentie_tabel")
@SequenceGenerator(schema = SCHEMA, name = "sq_humantask_referentie_tabel", sequenceName = "sq_humantask_referentie_tabel", allocationSize = 1)
public class HumanTaskReferentieTabel {

    @Id
    @GeneratedValue(generator = "sq_humantask_referentie_tabel", strategy = GenerationType.SEQUENCE)
    @Column(name = "id_humantask_referentie_tabel")
    private Long id;

    @NotNull @ManyToOne
    @JoinColumn(name = "id_referentie_tabel", referencedColumnName = "id_referentie_tabel")
    private ReferenceTable tabel;

    @NotNull @ManyToOne
    @JoinColumn(name = "id_humantask_parameters", referencedColumnName = "id_humantask_parameters")
    private HumanTaskParameters humantask;

    @NotBlank @Column(name = "veld", nullable = false)
    private String veld;

    public HumanTaskReferentieTabel() {
    }

    public HumanTaskReferentieTabel(final String veld, final ReferenceTable tabel) {
        this.veld = veld;
        this.tabel = tabel;
    }

    public Long getId() {
        return id;
    }

    public void setId(final Long id) {
        this.id = id;
    }

    public ReferenceTable getTabel() {
        return tabel;
    }

    public void setTabel(final ReferenceTable tabel) {
        this.tabel = tabel;
    }

    public HumanTaskParameters getHumantask() {
        return humantask;
    }

    public void setHumantask(final HumanTaskParameters humantask) {
        this.humantask = humantask;
    }

    public String getVeld() {
        return veld;
    }

    public void setVeld(final String veld) {
        this.veld = veld;
    }
}

File: src/main/kotlin/nl/info/zac/admin/model/HumanTaskReferentieTabel.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.zac.admin.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.SequenceGenerator
import jakarta.persistence.Table
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import nl.info.zac.database.flyway.FlywayIntegrator.Companion.SCHEMA
import nl.info.zac.util.AllOpen

@Entity
@Table(schema = SCHEMA, name = "humantask_referentie_tabel")
@SequenceGenerator(
    schema = SCHEMA,
    name = "sq_humantask_referentie_tabel",
    sequenceName = "sq_humantask_referentie_tabel",
    allocationSize = 1
)
@AllOpen
class HumanTaskReferentieTabel() {

    constructor(veld: String, tabel: ReferenceTable) : this() {
        this.veld = veld
        this.tabel = tabel
    }

    @Id
    @GeneratedValue(generator = "sq_humantask_referentie_tabel", strategy = GenerationType.SEQUENCE)
    @Column(name = "id_humantask_referentie_tabel")
    var id: Long? = null

    @field:NotNull
    @ManyToOne
    @JoinColumn(name = "id_referentie_tabel", referencedColumnName = "id_referentie_tabel")
    var tabel: ReferenceTable? = null

    @field:NotNull
    @ManyToOne
    @JoinColumn(name = "id_humantask_parameters", referencedColumnName = "id")
    var humantask: ZaaktypeCmmnHumantaskParameters? = null

    @field:NotBlank
    @Column(name = "veld", nullable = false)
    var veld: String? = null

    override fun equals(other: Any?): Boolean {
        if (other !is HumanTaskReferentieTabel) return false
        return tabel == other.tabel && veld == other.veld
    }

    override fun hashCode(): Int {
        var result = tabel?.hashCode() ?: 0
        result = 31 * result + (veld?.hashCode() ?: 0)
        return result
    }
}
